testsuite/smoke-test/src/upgrade.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    smoke_test_environment::SwarmBuilder, test_utils::check_create_mint_transfer,
+    workspace_builder, workspace_builder::workspace_root,
+};
+use aptos_crypto::ValidCryptoMaterialStringExt;
+use aptos_gas::{AptosGasParameters, GasQuantity, InitialGasSchedule, ToOnChainGasSchedule};
+use aptos_temppath::TempPath;
+use forge::Swarm;
+use std::process::Command;
+use std::{fmt::Write, fs};
+
+fn generate_blob(data: &[u8]) -> String {
+    let mut buf = String::new();
+
+    write!(buf, "vector[").unwrap();
+    for (i, b) in data.iter().enumerate() {
+        if i % 20 == 0 {
+            if i > 0 {
+                writeln!(buf).unwrap();
+            }
+        } else {
+            write!(buf, " ").unwrap();
+        }
+        write!(buf, "{}u8,", b).unwrap();
+    }
+    write!(buf, "]").unwrap();
+    buf
+}
+
+#[tokio::test]
+/// This test verifies the flow of aptos framework upgrade process.
+/// i.e: The network will be alive after applying the new aptos framework release.
+async fn test_upgrade_flow() {
```

Comment:
this is only a partial upgrade, we have a lot of upgrades to do, see https://github.com/aptos-labs/aptos-genesis-waypoint/tree/main/testnet/proposals/sources/221007-complete-upgrade

we need to do
gas -> stdlib -> aptos stdlib -> aptos framework -> token -> enable any new features

Reply:
I'll add the full upgrade flow.